scala/private/common.bzl

```diff
@@ -166,12 +166,33 @@ def _provider_of_dependency_label_of(dependency, path):
 
 # TODO this seems to have limited value now that JavaInfo has everything
 def create_java_provider(scalaattr, transitive_compile_time_jars):
-    return java_common.create_provider(
-        use_ijar = False,
-        compile_time_jars = scalaattr.compile_jars,
-        runtime_jars = scalaattr.transitive_runtime_jars,
-        transitive_compile_time_jars = depset(
-            transitive = [transitive_compile_time_jars, scalaattr.compile_jars],
-        ),
-        transitive_runtime_jars = scalaattr.transitive_runtime_jars,
-    )
+    compile_time_deps = [
+        JavaInfo(
+            output_jar = jar,
+            compile_jar = jar,
+        )
+        #TODO: Avoid flattening the depset here.
+        for jar in transitive_compile_time_jars.to_list()
+    ]
+
+    compile_providers_neverlink = [
+        JavaInfo(
+            output_jar = jar,
+            compile_jar = jar,
+            # these are only header jars and they should not be used as runtime
+            neverlink = True)
```

Comment:
I think this is wrong. These aren't "compile only" but are derived from `deps` which means "compile" and "runtime". Now that you mention it I'm thinking we might have a bug in ignoring neverlink for some cases but I need to dive more into the code